src/kubernetes.cc

```diff
@@ -322,8 +324,34 @@ MetadataUpdater::ResourceMetadata KubernetesReader::GetContainerMetadata(
   const std::string k8s_container_name = boost::algorithm::join(
       std::vector<std::string>{kK8sContainerNameResourcePrefix, namespace_name, pod_name, container_name},
       kResourceTypeSeparator);
+
+  std::vector<std::string> local_resource_ids = {
+    k8s_container_pod,
+    k8s_container_name,
+  };
+
+  if (container_status) {
+    std::size_t docker_prefix_end = sizeof(kDockerIdPrefix) - 1;
```

Comment:
perhaps use strlen here (instead of sizeof) so you don't have to subtract by 1.

Also, docker_prefix_end -> docker_prefix_len is easier to parse.

Reply:
`sizeof` makes it a compile-time constant... I'd declare this `constexpr`, anyway...

As for `end` vs `len`, "len" is an abbreviation, so maybe even "length" or "count" (see the [signature of `compare`](http://en.cppreference.com/w/cpp/string/basic_string/compare)). I used `end` because it's the position of the `'\0'`, which is the end of the C string...

Reply:
Should we go with `docker_prefix_length`? As for the `constexpr` request, are you just asking for me to prefix this line with that?

Reply:
`docker_prefix_length` SGTM. And yes, please prefix the variable declaration with `constexpr`.

Reply:
Updated, stayed away from constexpr due to a modification within the conditional statement.